app/plugin_commands_test.go

```diff
@@ -439,3 +443,188 @@ func TestPluginCommand(t *testing.T) {
 	})
 
 }
+
+// Test Product with the minimum code needed to handle
+// hooksmanager and slash commands
+type TProduct struct {
+	hooksService product.HooksService
+}
+
+func newTProduct(m map[product.ServiceKey]any) (product.Product, error) {
+	return &TProduct{
+		hooksService: m[product.HooksKey].(product.HooksService),
+	}, nil
+}
+func (p *TProduct) Start() error {
+	p.hooksService.RegisterHooks("productT", p)
+	return nil
+}
+func (p *TProduct) Stop() error { return nil }
+func (p *TProduct) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*model.CommandResponse, *model.AppError) {
+	return &model.CommandResponse{Text: "product slash command called"}, nil
+}
+
+func TestProductCommands(t *testing.T) {
+
+	products := map[string]product.Manifest{
+		"productT": {
+			Initializer:  newTProduct,
+			Dependencies: map[product.ServiceKey]struct{}{},
+		},
+	}
+
+	t.Run("Execute product command", func(t *testing.T) {
+		th := Setup(t).InitBasic()
+		defer th.TearDown()
+		// Server hijack.
+		// This must be done in a cleaner way.
+		th.Server.initializeProducts(products, th.Server.services)
```

Comment:
Let's add a TODO statement here if we agree deferring this to a later time. (with the ticket number)